File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int result = 0;
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int i:nums)
        {
            if(map.get(i)==null)
            {
                map.put(i,map.getOrDefault(i-1,0)+map.getOrDefault(i+1,0)+1);
                map.put(i-map.getOrDefault(i-1,0),map.get(i));
                map.put(i+map.getOrDefault(i+1,0),map.get(i));
                result = Math.max(result,map.get(i));
            }

        }
        return result;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int result=0;
        int start=0,end=0;
        map<char,int> map;
        while(end<s.length())
        {
            if(map[s[end]]==0)
            {
                map[s[end]]++;
                result = max(result,end-start+1);
                end++;
            }
            else
            {
                while(s[start]!=s[end])
                {
                    map[s[start]]--;
                    start++;
                }
                map[s[start]]--;
                start++;
            }
        }
        return result;
    }
};
